Library/Homebrew/cmd/config.rb

```diff
@@ -119,8 +119,10 @@ def describe_system_ruby
     s
   end
 
-  def java_version
-    java = `java -version 2>&1`.lines.first.chomp
+  def describe_java
+    java = which "java"
+    return "N/A" if java.nil?
+    java = `#{java} -version 2>&1`.lines.first.chomp
```

Comment:
Apologies for the delay in replying; Was rebooting and trying bash, and then rebooting and trying Terminal, and then rebooting and trying iTerm2, and so on. Wanted to get a pretty wide sample.

` /usr/libexec/java_home --exec java -version ` = 17-30 seconds on average.

` /usr/libexec/java_home ` = Less than a second.

Readouts respectively are:

```
java version "1.8.0_25"
Java(TM) SE Runtime Environment (build 1.8.0_25-b17)
Java HotSpot(TM) 64-Bit Server VM (build 25.25-b02, mixed mode)
```

and

` /Library/Java/JavaVirtualMachines/jdk1.8.0_25.jdk/Contents/Home `.